### DISCUSSION TITLE
Safelist pattern including dark modifier

### DISCUSSION THREAD

**Discussion by agualis** [tailwindlabs/tailwindcss#10079] [Help] [answered]
I have a [project](https://github.com/agualis/tailwind3-safelist) (tailwind 3.2 + vite + vue) where I have a [button](https://stackblitz.com/github/agualis/tailwind3-safelist?file=src%2FApp.vue) with a dynamically generated class like this one: 

`dark:hover:border-${color}-700`

where color is an interpolated variable with value `pink`.

Therefore, I want to include `dark:hover:border-pink-700` to my safelist with a pattern that allows other colors so that I use this configuration:

```javascript
// tailwind.config.js
...
safelist: [
    {
      pattern: /border-[^/]+$/,
      variants: ['dark', 'hover', 'focus'],
    },
  ],
...
```

Using `hover` and `focus` variants inside the previous pattern do work but, `dark` is not working so that: 

- `focus:border-pink-700` and similar light patterns are included in the final css. 
- `dark:focus:border-pink-700` and similar dark patterns are excluded from the final css. 

**Do I need a different setup for adding dark variant?** **Am I missing something?**

Here is an example where I can reproduce the problem: 

https://stackblitz.com/github/agualis/tailwind3-safelist?file=src%2FApp.vue

([linked repo](https://github.com/agualis/tailwind3-safelist)).

**Comment by wongjn** [comment, marked as answer]
You need to specify the combination variants, like:

```js
module.exports {
  // …
  safelist: [
    {
      pattern: /border-[^/]+$/,
      variants: [
        'dark',
        'hover',
        'focus',
        'dark:hover',
        'dark:focus',
      ,],
    },
  ],
  // …
};
...

**Reply by agualis** [reply]
Thanks @wongjn! It worked 💃

I pushed a branch with your suggested solution in case it is helpful for anyone:

[Working sample
](https://stackblitz.com/github/agualis/tailwind3-safelist/tree/fix?file=tailwind.config.js)

[Branch with fix](https://github.com/agualis/tailwind3-safelist/tree/fix)

This detail wasn't clear for me. Maybe a more concrete example in the docs could be useful for other people.
